cpp/daal/src/algorithms/low_order_moments/oneapi/low_order_moments_distributed_oneapi_impl.i

```diff
@@ -0,0 +1,488 @@
+                                                                                          PartialResult * partialResult, const Parameter * parameter)
+{
+    services::Status status;
+
+    auto & context = daal::services::internal::getDefaultContext();
+
+    if (method == defaultDense)
+    {
+        if (parameter->estimatesToCompute == estimatesMinMax)
+        {
+            LowOrderMomentsDistributedTaskOneAPI<algorithmFPType, estimatesMinMax> task(context, partialResultsCollection, partialResult, status);
+            DAAL_CHECK_STATUS_VAR(status);
+            return task.compute();
+        }
+        else if (parameter->estimatesToCompute == estimatesMeanVariance)
+        {
+            LowOrderMomentsDistributedTaskOneAPI<algorithmFPType, estimatesMeanVariance> task(context, partialResultsCollection, partialResult,
+                                                                                              status);
+            DAAL_CHECK_STATUS_VAR(status);
+            return task.compute();
+        }
+        else
+        {
+            /* estimatesAll */
+            LowOrderMomentsDistributedTaskOneAPI<algorithmFPType, estimatesAll> task(context, partialResultsCollection, partialResult, status);
+            DAAL_CHECK_STATUS_VAR(status);
+            return task.compute();
+        }
+    }
+
+    return status;
+}
+
+template <typename algorithmFPType, Method method>
+services::Status LowOrderMomentsDistributedKernelOneAPI<algorithmFPType, method>::finalizeCompute(PartialResult * partialResult, Result * result,
+                                                                                                  const Parameter * parameter)
+{
+    services::Status status;
+
+    auto & context = daal::services::internal::getDefaultContext();
+    if (method == defaultDense)
+    {
+        /*nothing is done in case of finalizing results which are already available in partialResult (i.e. min max ...),
+          due to they should be already assigned into result from partial results by level up caller
+        */
+        if (parameter->estimatesToCompute == estimatesMeanVariance)
+        {
+            LowOrderMomentsDistributedFinalizeTaskOneAPI<algorithmFPType, estimatesMeanVariance> task(context, partialResult, result, status);
+            DAAL_CHECK_STATUS_VAR(status);
+            return task.compute();
+        }
+        else if (parameter->estimatesToCompute == estimatesAll)
+        {
+            /* estimatesAll */
+            LowOrderMomentsDistributedFinalizeTaskOneAPI<algorithmFPType, estimatesAll> task(context, partialResult, result, status);
+            DAAL_CHECK_STATUS_VAR(status);
+            return task.compute();
+        }
+    }
+
+    return status;
+}
+
+template <typename T, typename Q, typename P>
+static inline services::Status overflowCheckByMultiplication(const Q & v1, const P & v2)
+{
+    DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(T, v1, v2);
+    return services::Status();
+}
```

Comment:
Looks strange at least. Calls of this function are coupled with checks. I believe, it can be done by raw `DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION` usage

Reply:
I need this function-wrapper to prevent direct return (performed by macros) in case of using it in constructor.